custom/icds_reports/sqldata/exports/dashboard_usage.py

```diff
@@ -0,0 +1,177 @@
+            if result[sub_location] not in location_ids:
+                location_ids.append(result[sub_location])
+        return location_matrix, location_ids
+
+    def get_location_id_string_from_location_type(self, location_type):
+        if location_type == 'awc':
+            location_type = 'doc_id'
+        else:
+            location_type = location_type + '_id'
+        return location_type
+
+    def check_if_date_in_last_week(self, date):
+        if date is None:
+            return False
+        d = datetime.datetime.strptime(date, "%Y-%m-%d")
+        now = datetime.datetime.now()
+        return (d - now).days < 7
+
+    def is_launched_from_location_id(self, location_type, location_id):
+        filter_dict = {location_type + '_id': location_id, 'is_launched': 'yes'}
+        return AggAwc.objects.filter(**filter_dict).exists()
+
+    def get_excel_data(self):
+        excel_rows = []
+        filters = [['Generated at', india_now()]]
+        headers = ['Sr.No', 'State/UT Name', 'District Name', 'Block Name', 'Username', 'Level', 'Role',
+                   'Launched?', 'Last Login', 'Logged in the last week?', 'Total', 'Child', 'Pregnant Women',
+                   'Demographics', 'System Usage', 'AWC infrastructure', 'Child Growth Monitoring List',
+                   'ICDS - CAS Monthly Register', 'AWW Performance Report', 'LS Performance Report',
+                   'Take Home Ration']
+        excel_rows.append(headers)
+        serial_count = 0
+        for user_location in self.user.get_sql_locations(self.domain):
+            # getting the location types to retrieve for this user location
+            location_type_filter = {
+                self.get_location_id_string_from_location_type(user_location.location_type_name):
+                    user_location.get_id}
+            all_awc_locations = AwcLocation.objects.filter(**location_type_filter).values(*self.required_fields)
+            # converting the result set to matrix to fetch ancestors for a given location
+            location_matrix, location_ids =\
+                self.convert_rs_to_matrix(all_awc_locations, self.get_location_id_string_from_location_type(
+                    user_location.location_type_name))
+            users = self.get_users_by_location(location_ids)
+            records = list(ICDSAuditEntryRecord.objects.filter(url='/a/icds-dashboard-qa/cas_export')
+                           .annotate(indicator=KeyTextTransform('indicator', 'post_data')).values('indicator',
+                                                                                                  'username')
+                           .annotate(count=Count('indicator')).order_by('username', 'indicator'))
+            # accumulating the indicator counts
+            for user in users:
+                indicator_count = []
+                total_indicators = 0
+                for record in records:
+                    if record['username'] == user['username']:
```

Comment:
assuming you will be changing this logic based on Cals above comment.

Reply:
done